### PR TITLE
Renaming of existing parameters

### PR DESCRIPTION
**Pull request by dbadura**
**Description**

Changes proposed in this pull request:
- remove all white/black existing parameters to be more easier to understand

**Related issue(s)**
#5

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -39,6 +39,9 @@ tidy:
 test:
 	go test ./...
 
+build:
```
Comment (tgorgol) [inline Makefile:43]: Shouldn't this be a PHONY target?

File: cli/cli.go [hunk 2]
```diff
@@ -81,7 +82,6 @@ func runCmd(cmd string, shell bool) []byte {
 		out, err := exec.Command("/bin/bash", "-c", cmd).Output()
 		if err != nil {
 			log.Fatal(err)
-			panic("some error found")
```
Comment (tgorgol) [inline cli/cli.go:84]: Are we sure we can delete this panic? Won't there be a situation where some command fails but the tool reports success?

File: README.md [hunk 3]
```diff
@@ -33,21 +33,21 @@ go build
...
-| `-request-repeats`   | number of request repeats                               | `1`                |
-| `-allow-code-blocks` | checking links in code blocks                           | `false`            |
-| `-timeout`           | connection timeout (in seconds)                         | `30`               |
-| `-ignore-external`   | ignore external links                                   | `false`            |
-| `-ignore-internal`   | ignore internal links                                   | `false`            |
-| `-v`                 | enable verbose logging                                  | `false`            |
-| `-help` or `-h`      | Show available parameters                               | n/a                |
+| Name                           | Description                                             | Default Value      |
+| ------------------------------ | ------------------------------------------------------- | ------------------ |
+| `-base-path`                   | root directory of repository                            | `""`               |
```
Comment (kazydek) [inline README.md]:
```suggestion
| `-base-path`                   | Root directory of the repository                            | `""`               |
```

File: README.md [hunk 4]
```diff
@@ -33,21 +33,21 @@ go build
...
+| `-config-file`                 | configuration file for bot. See [more](#config-file)    | `milv.config.yaml` |
```
Comment (kazydek) [inline README.md]:
```suggestion
| `-config-file`                 | Configuration file for the bot. See the [**Config file**](#config-file) section for more details.  | `milv.config.yaml` |
```

File: README.md [hunk 5]
```diff
@@ -33,21 +33,21 @@ go build
...
+| `-external-links-to-ignore`    | comma separate external links which will not be checked | `[]`               |
```
Comment (kazydek) [inline README.md]:
```suggestion
| `-external-links-to-ignore`    | Comma-separated external links which will not be checked | `[]`               |
```

File: README.md [hunk 6]
```diff
@@ -33,21 +33,21 @@ go build
...
+| `-internal-links-to-ignore`    | comma separate internal links which will not be checked | `[]`               |
```
Comment (kazydek) [inline README.md]:
```suggestion
| `-internal-links-to-ignore`    | Comma-separated internal links which will not be checked | `[]`               |
```

File: README.md [hunk 7]
```diff
@@ -33,21 +33,21 @@ go build
...
+| `-files-to-ignore`             | comma separate files which will not be checked          | `[]`               |
```
Comment (kazydek) [inline README.md]:
```suggestion
| `-files-to-ignore`             | Comma-separated files which will not be checked          | `[]`               |
```

File: README.md [hunk 8]
```diff
@@ -33,21 +33,21 @@ go build
...
+| `-allow-redirect`              | redirects will be allowed                               | `false`            |
+| `-request-repeats`             | number of request repeats                               | `1`                |
+| `-allow-code-blocks`           | checking links in code blocks                           | `false`            |
+| `-timeout`                     | connection timeout (in seconds)                         | `30`               |
+| `-ignore-external`             | ignore external links                                   | `false`            |
+| `-ignore-internal`             | ignore internal links                                   | `false`            |
+| `-v`                           | enable verbose logging                                  | `false`            |
+| `-help` or `-h`                | show available parameters                               | n/a                |
```
Comment (kazydek) [inline README.md]:
```suggestion
| `-help` or `-h`                | Available parameters                               | n/a                |
```

File: README.md [hunk 9]
```diff
@@ -33,21 +33,21 @@ go build
...
+| `-external-links-to-ignore`    | comma separate external links which will not be checked | `[]`               |
+| `-internal-links-to-ignore`    | comma separate internal links which will not be checked | `[]`               |
+| `-files-to-ignore`             | comma separate files which will not be checked          | `[]`               |
+| `-allow-redirect`              | redirects will be allowed                               | `false`            |
+| `-request-repeats`             | number of request repeats                               | `1`                |
+| `-allow-code-blocks`           | checking links in code blocks                           | `false`            |
+| `-timeout`                     | connection timeout (in seconds)                         | `30`               |
+| `-ignore-external`             | ignore external links                                   | `false`            |
+| `-ignore-internal`             | ignore internal links                                   | `false`            |
+| `-v`                           | enable verbose logging                                  | `false`            |
```
Comment (kazydek) [inline README.md]:
```suggestion
| `-v`                           | Verbose logging                                  | `false`            |
```

**Review by tgorgol — COMMENTED** [review]
Good job! Few comments from me.

File: cli/cli.go (hunk 2 above)
Comment (dbadura) [reply cli/cli.go:84]: log.Fatal close the program with exit code 1.

File: README.md [hunk 10]
```diff
@@ -33,21 +33,21 @@ go build
...
-| `-v`                 | enable verbose logging                                  | `false`            |
-| `-help` or `-h`      | Show available parameters                               | n/a                |
+| Name                           | Description                                             | Default Value      |
+| ------------------------------ | ------------------------------------------------------- | ------------------ |
+| `-base-path`                   | root directory of repository                            | `""`               |
+| `-config-file`                 | configuration file for bot. See [more](#config-file)    | `milv.config.yaml` |
+| `-external-links-to-ignore`    | comma separate external links which will not be checked | `[]`               |
+| `-internal-links-to-ignore`    | comma separate internal links which will not be checked | `[]`               |
+| `-files-to-ignore`             | comma separate files which will not be checked          | `[]`               |
+| `-allow-redirect`              | redirects will be allowed                               | `false`            |
```
Comment (kazydek) [inline README.md]:
```suggestion
| `-allow-redirect`              | Redirects will be allowed                               | `false`            |
```

File: README.md [hunk 11]
```diff
@@ -33,21 +33,21 @@ go build
...
+| `-request-repeats`             | number of request repeats                               | `1`                |
```
Comment (kazydek) [inline README.md]:
```suggestion
| `-request-repeats`             | Number of repeated request                               | `1`                |
```

File: README.md [hunk 12]
```diff
@@ -33,21 +33,21 @@ go build
...
+| `-allow-code-blocks`           | checking links in code blocks                           | `false`            |
```
Comment (kazydek) [inline README.md]:
```suggestion
| `-allow-code-blocks`           | Links in code blocks will be checked                          | `false`            |
```

File: README.md [hunk 13]
```diff
@@ -33,21 +33,21 @@ go build
...
+| `-timeout`                     | connection timeout (in seconds)                         | `30`               |
```
Comment (kazydek) [inline README.md]:
```suggestion
| `-timeout`                     | Connection timeout (in seconds)                         | `30`               |
```

File: README.md [hunk 14]
```diff
@@ -33,21 +33,21 @@ go build
...
+| `-ignore-external`             | ignore external links                                   | `false`            |
```
Comment (kazydek) [inline README.md]:
```suggestion
| `-ignore-external`             | External links to be ignored                                  | `false`            |
```

File: README.md [hunk 15]
```diff
@@ -33,21 +33,21 @@ go build
...
+| `-ignore-internal`             | ignore internal links                                   | `false`            |
```
Comment (kazydek) [inline README.md]:
```suggestion
| `-ignore-internal`             | Internal links to be ignored                                   | `false`            |
```

File: README.md [hunk 16]
```diff
@@ -159,15 +159,15 @@ The below table describes the types of errors during checking links and examples
...
 | `Get <external link>: EOF `                                                                        | Same as above or change the link to the other one (probably website doesn't exist)                                                                                                                                                                                            |
-| Other types of errors and errors with contains `no such host` or `timeout` words                   | Most likely, the website doesn't exist or you do not have access to it. Possible solutions: change the link to another, correct one, remove it or add it to the `white-list-external` or `white-list-internal`                                                                |
+| Other types of errors and errors with contains `no such host` or `timeout` words                   | Most likely, the website doesn't exist or you do not have access to it. Possible solutions: change the link to another, correct one, remove it or add it to the `external-links-to-ignore` or `internal-links-to-ignore`                                                                |
```
Comment (kazydek) [inline README.md]:
```suggestion
| Other types of errors and errors that contain the `no such host` or `timeout` words                   | It means that the website doesn't exist or you don't have access to it. You can change the link to another one, correct or remove it, or add it to the `external-links-to-ignore` or `internal-links-to-ignore` list.                                                               |
```

File: README.md [hunk 17]
```diff
@@ -159,15 +159,15 @@ The below table describes the types of errors during checking links and examples
...
 | `404 Not Found`                                                                                    | Page doesn't exist - you have to change the external link to the correct one                                                                                                                                                                                                  |
-| Error with formatting link                                                                         | Correct link or if link has a variables or it is a example, add this link to the `white-list-external` or `white-list-internal`                                                                                                                                               |
+| Error with formatting link                                                                         | Correct link or if link has a variables or it is a example, add this link to the `external-links-to-ignore` or `internal-links-to-ignore`                                                                                                                                               |
```
Comment (kazydek) [inline README.md]:
```suggestion
| Error with formatting the link                                                                         | Correct the link. If the link contains variables or is used as an example, add it to the `external-links-to-ignore` or `internal-links-to-ignore` list.                                                                                                                                              |
```

File: README.md [hunk 18]
```diff
@@ -99,23 +99,23 @@ If your tree of your project look like this:
...
   - path: "./src/foo.md"
     config:
-      white-list-external: ["github.com"]
-      white-list-internal: ["#contributing"]
+      external-links-to-ignore: ["github.com"]
+      internal-links-to-ignore: ["#contributing"]
 ```
 
-Before run validation, `milv` remove from files list `./README.md` file to check and connect global `white-list-external` with file `white-list-external` and `white-list-external` for `./src/foo.md` file will look that:
+Before run validation, `milv` remove from files list `./README.md` file to check and connect global `external-links-to-ignore` with file `external-links-to-ignore` and `external-links-to-ignore` for `./src/foo.md` file will look that:
```
Comment (kazydek) [inline README.md]:
```suggestion
Before running the validation, `milv` removes the `./README.md` file from the files list. It then concatenates the values for `external-links-to-ignore` entries. For the `./src/foo.md` file mentioned in the example, the result will look as follows:
```

**Review by kazydek — COMMENTED** [review]
The doc will be rewritten as a part of this follow-up issue: https://github.com/kyma-incubator/milv/issues/2

File: cli/cli.go (hunk 2 above)
Comment (tgorgol) [reply cli/cli.go:84]: True, missed that

**Review by tgorgol — APPROVED** [review]
LGTM, let's just make sure that the names we came up with are easy to understand for TWs before merging the PR.

**Review by kazydek — APPROVED** [review]

**Pull request merged**